lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -154,7 +154,7 @@ def _get(self, params: ApiParams, request: Request) -> Response:
         for dataset in datasets:
             d = {
                 "name": dataset.name,
-                "run_id": dataset.md5,
+                "resource_id": dataset.resource_id,
```

Comment:
Resources have an informational name as well as the formal resource ID; nobody wants to display the resource ID, and the name (whatever it's been set to) provides a more friendly display.

The default `dataset.name` will be `dataset.metalog.pbench.script` + `dataset.metalog.pbench.config` + `dataset.metalog.pbench.date` just as it is now, but I plan to make a change to `PUT /datasets/metadata` to allow changing it now that it's not essential internally. ([PBENCH-749](https://issues.redhat.com/browse/PBENCH-749))

So, yes, the default response is the resource ID and the resource name.

Reply:
This might be out of scope for this PR, but what are we thinking the Dashboard is going to do with this?  

My previous understanding was that we were expecting the Dashboard to display the results list with the script, config, and date in three separate columns, with the user able to sort on any of the columns.  If the dataset name is still a thing, then we have four columns, and the others are kind of redundant with the name (up until the user changes the name).

Having the default display be the name, and having the default name be `script`+`config`+`date` seems like we haven't improved anything, here.  On the other hand, having the Dashboard display `script`,`config`, and `date` (and _not_ `name`) seems like a nice improvement...right up until we allow the user to change the name, at which point we want to be able to display the name...but then that screws up what we display by default (I think).

So, I'm questioning whether we really want a name field at all.  We know what we would put in it by default if we had it, but that just makes it redundant with other stuff that we want to do.  We do want to enable users to do certain things, and one way to do that is to allow them to set the name, but I feel like there are other ways to enable that support which don't involve a _name_ field, per se.  I'm proposing that we provide "labels" or "annotations" and that we get rid of the `name` field altogether.  I believe that we've already got ways to display our equivalent of the "name" (via the three components) and that, if the user wants to create a custom handle for the result, they should do it via a label or annotation, and that delivers us from the question of whether/when to display the name versus the `script`/`config`/`date` trio.

Reply:
I guess what is stuck in my craw about this is that the default behavior doesn't land the user anywhere near where we want to be.  That is, we think that having a name for run results would be The Thing, but in order to get there, the user has to explicitly specify an option (that we don't support, yet).  Instead, the typical case is going to be the run results get the default name (with the `script`/`config`/`date` trio) which is the opposite of what we want (we want those three to be in separate columns).  And, yeah, the user can reconfigure his view to omit the `name` column and present the other three, but, since we think the run name is The Thing, that's probably not the view that we're going to offer by default.

Since the user has to take an explicit step to use the `--name` option, it seems like the user should have to take an explicit step to get the "name" column in the UI.  Then the default-taking user would have the `script`/`config`/`date` trio display by default, which I think would result in a much better experience.  There's nothing which prevents a user from getting The Thing, but s/he has to take explicit steps to get there; otherwise, the user gets the trio by default.